Public/Src/Cache/ContentStore/Distributed/Redis/RedisContentLocationStoreConfiguration.cs

```diff
@@ -116,6 +116,15 @@ public class RedisContentLocationStoreConfiguration : LocalLocationStoreConfigur
         /// </summary>
         public bool AreBlobsSupported => BlobExpiryTimeMinutes > 0 && MaxBlobCapacity > 0 && MaxBlobSize > 0;
 
+        /// <summary>
+        /// The number of consecutive redis connection errors that will trigger reconnection.
+        /// </summary>
+        /// <remarks>
+        /// Due to a bug in StackExchange.Redis, the client may fail to reconnect to Redis Azure Cache in some cases.
+        /// This configuration allows us to recreate a connection if there is no successful calls to redis and all the calls are failing with connectivity issues.
+        /// </remarks>
+        public int RedisConnectionErrorLimit { get; set; } = int.MaxValue;

```

Comment:
This isn't a big deal, but using MaxValue to almost-certainly disable instead of null to really truely disable seems likely to generate occasional mysterious resets if we roll this out and then decide to turn it off.

Reply:
Ok. switch to nullable.

---
In reply to: [347643577](https://github.com/microsoft/BuildXL/pull/1229#discussion_r347643577) [](ancestors = 347643577)